### PR TITLE
fix data type converting for num-types

### PR DESCRIPTION
**Pull request by tomsoftware**
- fix data type converting
- add new string (stringEx) data type... my s7 returns on read

### CODE REVIEW

**Comment by tomsoftware** [comment]
the original code uses: 
- byte to string to number converting
- byte shift by using pow(2, x)
this push uses
- byte to number converting by using bit operations

**Comment by mesta1** [comment]
Thanks, I'll have a look. Is there any open issue that are fixed in these commits ?

**Comment by tomsoftware** [comment]
- It fixes some float-point-problems in "Double.cs" for cases where the numbers a quiet small/big
- there were some cases with signed numbers where the result was wrong
- the new StringEx handels a String format my PLC returned on a read of string... don't know why

so may this fixes the  #95, #86 and #88 issue...

**Comment by mesta1** [comment]
I really need unit tests on this. It seems all okay and tests are green, but I'm not sure all things that you changed are covered by tests. I will have a closer look on the weekend, but if you can write some tests, it will speed the merge.

File: S7.Net/Types/Timer.cs [hunk 1]
```diff
@@ -13,26 +13,30 @@ public static class Timer
...
+            ///int unknown = (bytes[0] >> 6) & 0x03;
+
+            switch ((bytes[0] >> 4) & 0x03)
             {
-                case "00":
+                case 0:
                     wert *= 0.01;
                     break;
-                case "01":
+                case 1:
```
Comment (mbalous) [inline S7.Net/Types/Timer.cs]: Using binary literals might be more obvious 

```csharp
case 0b00:
    break;
case 0b01:
    break;
case 0b10:
    break;
case 0b11:
    break;
```

Comment (tomsoftware) [reply S7.Net/Types/Timer.cs]: Bit values are good for displaying FLAGs where every bit has a different meaning but in this case it is more or less a ENUM or even a exponentiation: ( wert *= 10^(n-2) ) so Int values are good I think?!

File: S7.Net/Types/DWord.cs [hunk 2]
```diff
@@ -12,36 +12,39 @@ public static class DWord
...
+            return (UInt32)(bytes[0] << 24 | bytes[1] << 16 | bytes[2] << 8 | bytes[3]);
         }
 
+
         /// <summary>
-        /// Converts a S7 DWord (4 bytes) to uint (UInt32)
+        /// Converts 4 bytes to DWord (UInt32)
         /// </summary>
-        public static UInt32 FromBytes(byte v1, byte v2, byte v3, byte v4)
+        public static UInt32 FromBytes(byte b1, byte b2, byte b3, byte b4)
```
Comment (rapha-dev) [inline S7.Net/Types/DWord.cs]: Maybe more obvious naming is better when talking about bytes in a DWord/Word, starting at byte 0.
`public static UInt32 FromBytes(byte b0, byte b1, byte b2, byte b3)`

**Pull request merged**